src/darwin/Framework/CHIP/MTRBaseDevice.mm

```diff
@@ -1470,6 +1471,327 @@ + (id)CHIPEncodeAndDecodeNSObject:(id)object
     return decodedData.GetDecodedObject();
 }
 
+template <typename DecodableEventType> class BufferedReadEventCallback final : public app::ReadClient::Callback {
```

Comment:
Just to repeat again: this is duplicating a _lot_ of the attribute code.  Why are we not sharing that code, after a bit of refactoring, instead of copying it?

Reply:
Added BufferedReadClientCallback instead of BufferedReadAttributeCallback and BufferedReadEventCallback to reduce duplication.